engine/src/flutter/impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -50,496 +52,581 @@ class PositionWriter {
-      if (!is_end_of_component) {
-        constexpr Scalar kAngleThreshold = 10 * kPi / 180;
-        // `std::cosf` is not constexpr-able, unfortunately, so we have to bake
-        // the alignment constant.
-        constexpr Scalar kAlignmentThreshold =
-            0.984807753012208;  // std::cosf(kThresholdAngle) -- 10 degrees
-
-        // Use a cheap dot product to determine whether the angle is too sharp.
-        if (previous_offset.GetAlignment(offset) < kAlignmentThreshold) {
-          Scalar angle_total = previous_offset.AngleTo(offset).radians;
-          Scalar angle = kAngleThreshold;
-
-          // Bridge the large angle with additional geometry at
-          // `kAngleThreshold` interval.
-          while (angle < std::abs(angle_total)) {
-            Scalar signed_angle = angle_total < 0 ? -angle : angle;
-            Point offset =
-                previous_offset.GetVector().Rotate(Radians(signed_angle));
-            vtx.position = polyline.GetPoint(point_i) + offset;
-            vtx_builder.AppendVertex(vtx.position);
-            vtx.position = polyline.GetPoint(point_i) - offset;
-            vtx_builder.AppendVertex(vtx.position);
-
-            angle += kAngleThreshold;
-          }
-        }
-      }
-
-      // For curve components, the polyline is detailed enough such that
-      // it can avoid worrying about joins altogether.
-      if (is_end_of_component) {
-        // Append two additional vertices to close off the component. If we're
-        // on the _last_ component of the contour then we need to use the
-        // contour's end direction.
-        // `ComputeOffset` returns the contour's end direction when attempting
-        // to grab offsets past `contour_end_point_i`, so just use `offset` when
-        // we're on the last component.
-        Point last_component_offset = is_last_component
-                                          ? offset.GetVector()
-                                          : previous_offset.GetVector();
-        vtx.position = polyline.GetPoint(point_i + 1) + last_component_offset;
-        vtx_builder.AppendVertex(vtx.position);
-        vtx.position = polyline.GetPoint(point_i + 1) - last_component_offset;
-        vtx_builder.AppendVertex(vtx.position);
-        // Generate join from the current line to the next line.
-        if (!is_last_component) {
-          join_proc(vtx_builder, polyline.GetPoint(point_i + 1),
-                    previous_offset.GetVector(), offset.GetVector(),
-                    scaled_miter_limit, scale);
-        }
-      }
+  static Scalar ComputeMinimumMiterCosine(Scalar miter_limit) {
+    if (miter_limit <= 1.0f) {
+      // Miter limits less than 1.0 are impossible to meet since the miter
+      // join will always be at least as long as half the line width, so they
+      // essentially eliminate all miters. We return a degenerate cosine
+      // value so that the join routine never adds a miter.
+      return 1.1f;
```

Comment:
Would scaling the path affect this?

Reply:
It's a cosine value. The valid range is [-1, 1]

Reply:
Also, I point out in two places (the impl comments and the new doc comment) that the miter calculations are scale independent because it is a ratio between 2 different values that scale together.